research/2022-07-15-relay-anonymity.md

```diff
@@ -0,0 +1,373 @@
+* (2) flooding neighbours with bogus messages
+
+While (1) has a negative effect on availability, the impact is not significant.
+A linearly scaling botnet attacker, however, could significantly disrupt the network with such an attack.
+(2) is thwarted by [RLN](https://rfc.vac.dev/spec/17/).
+Also [SWAP](https://rfc.vac.dev/spec/18/) helps mitigating DoS attacks.
+
+A local attacker can DoS Waku by dropping all Waku traffic within its controlled network segment.
+An AS attacker can DoS Waku within its authority, while a global attacker can DoS the whole network.
+A countermeasure are censorship resistance techniques like [Pluggable Transports](https://www.pluggabletransports.info/about/).
+
+## Summary and Future Work
+
+Currently, Waku v2 relay offers k-anonymity with respect to receiver anonymity.
+This also includes k-anonymity towards legitimate members of the same topic.
+
+Waku v2 relay offers sender anonymity in the single node attacker model with its [strict no sign policy](https://rfc.vac.dev/spec/11/#signature-policy).
+Currently, Waku v2 does not guarantee sender anonymity in the multi node and stronger attacker models.
+However, we are working on modular anonymity-preserving protocols and building blocks as part of our privacy/anonymity roadmap.
+The goal is to allow tunable anonymity with respect to trade offs between *strong anonymity*, *low bandwidth*, and *low latency*.
+All of these cannot be fully guaranteed as the [the anonymity trilemma](https://freedom.cs.purdue.edu/projects/trilemma.html) states.
+Some applications have specific requirements, e.g. low latency, which require a compromise on anonymity.
+Anonymity-preserving mechanisms we plan to investigate and eventually specify as pluggable anonymity protocols for Waku comprise
+
+* [Dandilion++](https://arxiv.org/abs/1805.11060) for lightweight anonymity;
+* [onion routing](https://en.wikipedia.org/wiki/Onion_routing) as a building block adding a low latency anonymization layer;
+* [a mix network](https://en.wikipedia.org/wiki/Mix_network) for providing strong anonymity (on top of onion routing) even in the strongest attacker model at the cost of higher latency.
+
+These pluggable anonymity-preserving protocols will form a sub-set of the Waku v2 protocol set.
+As an intermediate step, we might directly employ Tor for onion-routing, and [Nym](https://nymtech.net/) as a mix-net layer.
+
+In future research log posts, we will cover further Waku v2 protocols and identify anonymity problems that will be added to our roadmap.
+These protocols comprise
+
+* [13/WAKU2-STORE](https://rfc.vac.dev/spec/13/), which can violate receiver anonymity as it allows filtering by content topic.
+  A countermeasure is using the content topic exclusively for local filters.
+* [12/WAKU2-FILTER](https://rfc.vac.dev/spec/12/), which discloses nodes' interest in topics;
+* [19/WAKU2-LIGHTPUSH](https://rfc.vac.dev/spec/19/), which also discloses nodes' interest in topics;
```

Comment:
and links the lightpush client as the sender of a message to the lightpush service node

Reply:
5667897